category_encoders/quantile_encoder.py

```diff
@@ -0,0 +1,291 @@
+                estimate.loc[-2] = prior
+
+            mapping[col] = estimate
+
+        return mapping
+
+    def transform(self, X, y=None, override_return_df=False):
+        """Perform the transformation to new categorical data.
+        Parameters
+        ----------
+        X : array-like, shape = [n_samples, n_features]
+        y : array-like, shape = [n_samples] when transform by leave one out
+            None, when transform without target info (such as transform test set)
+        Returns
+        -------
+        p : array, shape = [n_samples, n_numeric + N]
+            Transformed values with encoding applied.
+        """
+
+        if self.handle_missing == "error":
+            if X[self.cols].isnull().any().any():
+                raise ValueError("Columns to be encoded can not contain null")
+
+        if self._dim is None:
+            raise ValueError(
+                "Must train encoder before it can be used to transform data."
+            )
+
+        # unite the input into pandas types
+        X = util.convert_input(X)
+
+        # then make sure that it is the right size
+        if X.shape[1] != self._dim:
+            raise ValueError(
+                "Unexpected input dimension %d, expected %d" % (X.shape[1], self._dim,)
+            )
+
+        # if we are encoding the training data, we have to check the target
+        if y is not None:
+            y = util.convert_input_vector(y, X.index)
+            if X.shape[0] != y.shape[0]:
+                raise ValueError(
+                    "The length of X is "
+                    + str(X.shape[0])
+                    + " but length of y is "
+                    + str(y.shape[0])
+                    + "."
+                )
+
+        if not list(self.cols):
+            return X
+
+        X = self.ordinal_encoder.transform(X)
+
+        if self.handle_unknown == "error":
+            if X[self.cols].isin([-1]).any().any():
+                raise ValueError("Unexpected categories found in dataframe")
+
+        X = self.quantile_encode(X)
+
+        if self.drop_invariant:
+            for col in self.drop_cols:
+                X.drop(col, 1, inplace=True)
+
+        if self.return_df or override_return_df:
+            return X
+        else:
+            return X.values
+
+    def quantile_encode(self, X_in):
+        X = X_in.copy(deep=True)
+
+        for col in self.cols:
+            X[col] = X[col].map(self.mapping[col])
+
+        return X
+
+    def get_feature_names(self):
+        """
+        Returns the names of all transformed / added columns.
+        Returns
+        -------
+        feature_names: list
+            A list with all feature names transformed or added.
+            Note: potentially dropped features are not included!
+        """
+
+        if not isinstance(self.feature_names, list):
+            raise ValueError(
+                "Must fit data first. Affected feature names are not known " "before."
```

Comment:
Is there a reason not to write `"... known before"` instead of `"... known " "before"`